pandas/io/formats/style.py

```diff
@@ -1389,26 +1405,22 @@ def _background_gradient(
         text_color_threshold: float = 0.408,
         vmin: Optional[float] = None,
         vmax: Optional[float] = None,
+        gmap: Optional[Sequence] = None,
     ):
         """
-        Color background in a range according to the data.
+        Color background in a range according to the data or a gradient map
         """
-        if (
-            not isinstance(text_color_threshold, (float, int))
-            or not 0 <= text_color_threshold <= 1
-        ):
-            msg = "`text_color_threshold` must be a value from 0 to 1."
-            raise ValueError(msg)
-
+        if gmap is None:
+            gmap = s.to_numpy(dtype=float)
+        else:
+            gmap = np.asarray(gmap, dtype=float).reshape(s.shape)
```

Comment:
you need some valiation on gmap here, its supposed to be an identical DataFrame right?

Reply:
its supposed to align with the axis, i.e. axis=0,1, must be one-d length of cols/rows, or axis=None then size of the dataframe input.
It will throw an error complaining about shape, albeit a rather generic one currently, if sizes are bad.